x-pack/plugins/stack_connectors/common/openai/constants.ts

```diff
@@ -18,6 +18,7 @@ export enum SUB_ACTION {
   RUN = 'run',
   INVOKE_AI = 'invokeAI',
   INVOKE_STREAM = 'invokeStream',
+  INVOKE_ASYNC_ITERATOR = 'invokeAsyncIterator',
```

Comment:
out of interest, why do we have these subactions (`invokeAI`, `invokeStream`, `invokeAsyncIterator`)?

Reply:
`invokeAsyncIterator` uses the [OpenAI node sdk](https://github.com/openai/openai-node), which is required by LangChain.

`invokeAI` and `invokeStream` use the OpenAI Rest API. I want to refactor these to use the OpenAI node sdk as well because I believe it will reduce risk. That would be for 8.15 What do you think?

Reply:
To be clear, after the refactor, we would only have `invokeAI` and `invokeStream` and they will both use OpenAI node sdk. `invokeAsyncIterator` would become `invokeStream`. One method for streaming, one method for non-streaming. Also open to renaming if you have better suggestion.

Another refactor for both GenAI connector types, I would like to create a new class, `GenAIConnector`, which would be subclass of `SubActionConnector`, just like [`CaseConnector`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/actions/server/sub_action_framework/README.md#classes) that would enforce the subaction method structure.

Reply:
@stephmilovic both are fine, whatever works with you! just trying to understand. I do think we'll want to support any URL for OpenAI because people use their own models with that option, but we can also add a third option for that.

Reply:
Added the url to both azure/openai OpenAI sdk paths so the url from the connector will be respected!